libusb/libusb.h

```diff
@@ -45,6 +45,8 @@ typedef SSIZE_T ssize_t;
 
 #if defined(__STDC_VERSION__) && (__STDC_VERSION__ >= 199901L)
 #define ZERO_SIZED_ARRAY		/* [] - valid C99 code */
+#elif defined(__WATCOMC__)
+#define ZERO_SIZED_ARRAY	1	/* [0] - non-standard, but usually working code */
```

Comment:
Have you checked that specifiying a one element array will not break existing code ? As far as I remember structure sizes are computed expecting arrays of having zero length. Your code change breaks that assumption.

Reply:
You are correct that the structure size changes with `[1]` instead of `[0]`.
I guess using libusb.h with watcom can only be possible by using its
undocumented `-za99` switch: Please ignore that part of the patch.

push/pop pragma parts of the patch are still needed, though.

Reply:
I guess the remaining change to enforce byte packing is also not sufficient. There are a whole lot of structures in libusb.h that require byte packing (effectively, that applies to all descriptor structures). But admittedly, this is also not properly defined for __GNU__ either.

Reply:
it only makes it behave like msvc.

Reply:
if you want to have the patch included, please add a new diff, which only has the parts needed.
In theory you could even add a pull request to the original libusb, which is on github as well.

Reply:
I already submitted to libusb mainstream and the patch is already in. https://github.com/libusb/libusb/commit/a2b81aeff1178a1c186986b1c608247b44b657f0

Close this one if needed.